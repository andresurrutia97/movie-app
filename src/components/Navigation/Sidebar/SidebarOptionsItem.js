import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const SidebarOptionsItem = ({ path, name }) => {
  return (
    <NavLink
      activeClassName="active"
      className="flex items-center"
      to={path}
      exact
    >
      <div className="flag" />
      <span className="ml-5">{name}</span>
    </NavLink>
  );
};

SidebarOptionsItem.propTypes = {
  path: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default SidebarOptionsItem;
