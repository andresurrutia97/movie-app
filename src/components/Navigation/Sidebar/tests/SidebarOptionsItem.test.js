import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SidebarOptionsItem from '../SidebarOptionsItem';

describe('<SidebarOptionsItem />', () => {
  const component = (
    <BrowserRouter>
      <SidebarOptionsItem path="/home" name="Home" />
    </BrowserRouter>
  );

  it('should render correctly', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have a path prop equal to "/home", a name prop equal to "Home"', () => {
    const wrapper = shallow(component);
    expect(wrapper.props().children.props.path).toEqual('/home');
    expect(wrapper.props().children.props.name).toEqual('Home');
  });
});
