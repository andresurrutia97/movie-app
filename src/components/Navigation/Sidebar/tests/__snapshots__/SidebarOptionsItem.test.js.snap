// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarOptionsItem /> should render correctly 1`] = `
<a
  aria-current={null}
  className="flex items-center"
  href="/home"
  onClick={[Function]}
>
  <div
    className="flag"
  />
  <span
    className="ml-5"
  >
    Home
  </span>
</a>
`;
